source/common/upstream/upstream_impl.h

```diff
@@ -304,7 +303,13 @@ class HostImpl : public HostDescriptionImpl,
 
   std::atomic<uint32_t> health_flags_{};
   std::atomic<uint32_t> weight_;
-  std::atomic<bool> used_;
+
+  struct HostHandleImpl : HostHandle {
+    HostHandleImpl(const HostImpl& parent) : parent_(parent) { parent.handle_count_++; }
+    ~HostHandleImpl() override { parent_.handle_count_--; }
+    const HostImpl& parent_;
```

Comment:
This is probably fine given the use, but would it be better to store a weak_ptr to the real host to avoid use after free?

Reply:
Ok.